rust/tools/authorization-logic/BUILD

```diff
@@ -134,11 +134,14 @@ filegroup(
     srcs = glob(["raksha_examples/**"]),
 )
 
-# Specifing `@souffle` as part of `data` in the `authorization_logic_test` target
+# Specifying `@souffle` as part of `data` in the `authorization_logic_test` target
 # triggers a build of souffle with `-fno-exceptions`, which causes the build
 # to fail. Instead, we extract the souffle binary using a genrule by making
 # it part of the `tools` (which does not trigger a rebuild) and use it when
 # running the authorization logic tests via `authorization_logic_test`.
+#
+# TODO(wkorman): Investigate warning as below on build:
```

Comment:
Can we file an issue and use that number in TODO? I think there is very straightforward fix, we should simply change the rule name to something like below:

```
genrule(
  name = "souffle_tool",
  outs = ["souffle"],
   ...
)
```

We will need to change the current uses of souffle though.

Reply:
Yes, that's probably the right solution. And you are correct, we would need to update our usages of `souffle`.

Reply:
Done, filed #607